kernel/packages/shared/atlas/sagas.ts

```diff
@@ -107,32 +117,51 @@ export function* checkAndReportAround() {
 }
 
 export function* reportScenesAround() {
-  const userPosition = lastPlayerPosition
   let atlasState = (yield select(state => state.atlas)) as AtlasState
   while (!atlasState.marketName['0,0']) {
     yield take(MARKET_DATA)
     atlasState = yield select(state => state.atlas)
   }
-  const data = atlasState.marketName
-  const targets: Record<string, MarketEntry> = {}
+
+  const userPosition = lastPlayerPosition
   const MAX_SCENES_AROUND = 15
   const userX = userPosition.x / parcelLimits.parcelSize
   const userY = userPosition.z / parcelLimits.parcelSize
+  const targets = getScenesAround({ x: userX, y: userY }, MAX_SCENES_AROUND, atlasState)
+
+  yield put(reportedScenes(Object.keys(targets), { x: userPosition.x, y: userPosition.z }))
+  yield call(reportScenes, atlasState, targets)
+}
+
+export function* reportScenesAroundParcel(action: ReportScenesAroundParcel) {
+  let atlasState = (yield select(state => state.atlas)) as AtlasState
+  while (!atlasState.marketName['0,0']) {
+    yield take(MARKET_DATA)
+    atlasState = yield select(state => state.atlas)
+  }
+
```

Comment:
Could we use the selector `getNameFromAtlasState` from `shared/atlas/selectors.ts` maybe? If not, please refactor to some `isMarketDataInitialized` selector.